src/model/document.js

```diff
@@ -391,6 +396,28 @@ export default class Document {
 	 * @event change
 	 * @param {module:engine/model/batch~Batch} batch The batch that was used in the executed changes block.
 	 */
+
+	/**
+	 * Fired after each {@link module:engine/model/model~Model#enqueueChange `enqueueChange()` block},
+	 * which changes the data model or the outermost {@link module:engine/model/model~Model#change `change()` block}
+	 * was executed and the document has been changed during that block's execution.
+	 *
+	 * This event fires before the {@link module:engine/model/model~Model#change} event.
+	 *
+	 * The changes which this event will cover include:
+	 *
+	 * * document structure changes,
+	 * * marker changes, which affects the data model.
+	 *
+	 * If you want to be notified about the document's model changes, then simply listen to this event like this:
```

Comment:
"document's model changes" makes no sense. If anything, this is "model's document".  Still, it doesn't make sense because you don't use this event to listen to document changes, but to listen to changes which affect the data. So to "data changes".